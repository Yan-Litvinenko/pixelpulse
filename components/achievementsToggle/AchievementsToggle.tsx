'use client';

import React from 'react';
import styles from '@/styles/components/achievementsToggle/AchievementsToggle.module.scss';
import type { AchievementsToggleProps } from '@/interface/achievements/achievements.interface';

export default function AchievementsToggle(props: AchievementsToggleProps): React.JSX.Element {
    const { checked, id, handleChange, textContent, value } = props;

    return (
        <label className={styles.label} htmlFor={id}>
            <input
                checked={checked}
                className={styles.label__input}
                id={id}
                name="achievements"
                onChange={handleChange}
                type="radio"
                value={value}
            />
            <div className={styles.label__checkbox}></div>
            <div className={styles.label__text}>{textContent}</div>
        </label>
    );
}
