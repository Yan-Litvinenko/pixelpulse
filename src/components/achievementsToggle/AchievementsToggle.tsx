import React from 'react';
import styles from './AchievementsToggle.module.scss';
import type { IAchievementsToggle } from '../../interfaces/interface.achievements';

const AchievementsToggle = (props: IAchievementsToggle): React.JSX.Element => {
    const { checked, id, onChange, textContent, value } = props;

    return (
        <label className={styles.label} htmlFor={id} data-testid="test-label">
            <input
                checked={checked}
                className={styles.label__input}
                id={id}
                name="achievements"
                onChange={onChange}
                type="radio"
                value={value}
            />
            <div className={styles.label__checkbox}></div>
            <div className={styles.label__text}>{textContent}</div>
        </label>
    );
};

export { AchievementsToggle };
