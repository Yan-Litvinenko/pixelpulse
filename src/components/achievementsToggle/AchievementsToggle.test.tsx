import React from 'react';
import { AchievementsToggle } from './AchievementsToggle';
import { render, screen } from '@testing-library/react';
import type { IAchievementsToggle } from '../../interfaces/interface.achievements';

const props: IAchievementsToggle = {
    checked: true,
    id: 'all',
    onChange: jest.fn(),
    textContent: 'test text',
    value: 'all',
};

describe('AchievementsToggle component', (): void => {
    test('Render with props', (): void => {
        render(<AchievementsToggle {...props} />);

        expect(screen.getByText(props.textContent)).toBeInTheDocument();
        expect(screen.getByTestId('test-achievements-label')).toMatchSnapshot();
    });
});
