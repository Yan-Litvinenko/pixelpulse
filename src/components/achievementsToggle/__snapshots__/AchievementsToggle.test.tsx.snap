// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AchievementsToggle component Render with props 1`] = `
<label
  data-testid="test-achievements-label"
  for="all"
>
  <input
    checked=""
    id="all"
    name="achievements"
    type="radio"
    value="all"
  />
  <div />
  <div>
    test text
  </div>
</label>
`;
